build/Makefile.gcov

```diff
@@ -1,35 +1,42 @@
 #
-# LCOV
+# GCOV
 #
 
-lcov: lcov-html
+GCOV_EXCLUDES = \
+	'$(top_srcdir)/ext/bcmath/libbcmath/*' \
+	'$(top_srcdir)/ext/date/lib/*' \
+	'$(top_srcdir)/ext/fileinfo/libmagic/*' \
+	'$(top_srcdir)/ext/gd/libgd/*' \
+	'$(top_srcdir)/ext/hash/sha3/*' \
+	'$(top_srcdir)/ext/mbstring/libmbfl/*' \
+	'$(top_srcdir)/ext/opcache/jit/libudis86/*' \
+	'$(top_srcdir)/ext/pcre/pcre2lib/*' \
+	'$(top_srcdir)/ext/xmlrpc/libxmlrpc/*'
 
-lcov-test: lcov-clean-data test
-
-php_lcov.info: lcov-test
+php_lcov.info: test
 	@echo "Generating lcov data for $@"
-	@$(LTP) --capture --no-external --directory . --output-file $@
+	$(LTP) --capture --no-external --directory . --output-file $@
 	@echo "Stripping bundled libraries from $@"
-	@$(LTP) --remove $@ \
-		'*/<stdout>' \
-		'$(top_srcdir)/ext/bcmath/libbcmath/*' \
-		'$(top_srcdir)/ext/date/lib/*' \
-		'$(top_srcdir)/ext/fileinfo/libmagic/*' \
-		'$(top_srcdir)/ext/gd/libgd/*' \
-		'$(top_srcdir)/ext/hash/sha3/*' \
-		'$(top_srcdir)/ext/mbstring/libmbfl/*' \
-		'$(top_srcdir)/ext/opcache/jit/libudis86/*' \
-		'$(top_srcdir)/ext/pcre/pcre2lib/*' \
-		'$(top_srcdir)/ext/xmlrpc/libxmlrpc/*' \
-		--output-file $@
+	$(LTP) --output-file $@ --remove $@ '*/<stdout>' $(GCOV_EXCLUDES)
 
 lcov-html: php_lcov.info
 	@echo "Generating lcov HTML"
-	@$(LTP_GENHTML) --legend --output-directory lcov_html/ --title "PHP Code Coverage" php_lcov.info
+	$(LTP_GENHTML) --legend --output-directory lcov_html/ --title "PHP Code Coverage" php_lcov.info
 
 lcov-clean:
 	rm -f php_lcov.info
 	rm -rf lcov_html/
 
 lcov-clean-data:
 	@find . -name \*.gcda -o -name \*.da -o -name \*.bbg? | xargs rm -f
+
+gcovr-html: test
```

Comment:
If we drop the `test` dependency here, then we could use those targets in the azure build. I think it makes sense to require the `test` step to be performed independently, because it usually requires configuration...

Reply:
Yes. Multiple targets can be run when needed: `make test gcovr-html`. 

You might have noticed that I removed the `lcov-clean-data` from the `php_lcov.info` target. The reason was because it causes an error when running multiple targets for example `make test lcov-html` fails because after the tests are run, the target `lcov-html` deletes the coverage data files and because the `test` target has already run it quite rightly doesn't run again.

There's a lot of duplication in the CI configurations. Using the makefile more would help simplify them. 

I've also been planning to cleanup the run-tests script a little. I've been working on documenting the usage and making the API more consistent.